File: api/src/models/room.py
from datetime import datetime
from pydantic import BaseModel

class Room(BaseModel):
    room_id: int
    room_code: str
    created_by: int
    name: str

class RoomCreateRequest(BaseModel):
    name: str
    created_by: int

class RoomUpdateRequest(BaseModel):
    name: str
    room_id: int
    
class RoomDeleteRequest(BaseModel):
    room_id: int
    membership_id: int
    is_admin: bool

File: client/models/Room.ts
export interface Room {
  roomId: number;
  roomCode: string;
  createdBy: number;
  name: string;
  createdAt: string;
  updatedAt: string;
}

export interface RoomCreateRequest {
  name: string;
  createdBy: number;
}

export interface RoomCreateResponse {
  roomId: number;
  roomCode: string;
  membershipId: number;
  role: string;
}

export interface RoomUpdateRequest {
  roomId: number;
  name: string;
}

export interface RoomDeleteRequest {
  roomId: number;
  membershipId: number;
  isAdmin: boolean;
}
